ReactWindows/ReactNative/ReactInstanceManagerImpl.cs

```diff
@@ -0,0 +1,238 @@
+            using (Tracer.Trace(Tracer.TRACE_TAG_REACT_BRIDGE, "buildJSModuleConfig"))
+            {
+                javaScriptModulesConfig = jsModulesBuilder.Build();
+            }
+
+            var javascriptRuntime = new CatalystInstance.Builder
+            {
+                QueueConfigurationSpec = CatalystQueueConfigurationSpec.Default,
+                JavaScriptExecutor = jsExecutor,
+                Registry = nativeModuleRegistry,
+                JavaScriptModulesConfig = javaScriptModulesConfig,
+                BundleLoader = jsBundleLoader,
+                NativeModuleCallExceptionHandler = ex => { } /* TODO */,
+            }.Build();
+
+            reactContext.InitializeWithInstance(javascriptRuntime);
+
+            await javascriptRuntime.InitializeBridgeAsync();
+            
+            return var;
+        }
+
+        private void ProcessPackage(
+            IReactPackage reactPackage,
+            ReactApplicationContext reactContext,
+            NativeModuleRegistry.Builder nativeRegistryBuilder,
+            JavaScriptModulesConfig.Builder jsModulesBuilder)
+        {
+            foreach (var nativeModule in reactPackage.CreateNativeModules(reactContext))
+            {
+                nativeRegistryBuilder.Add(nativeModule);
+            }
+
+            foreach (var type in reactPackage.CreateJavaScriptModulesConfig())
+            {
+                jsModulesBuilder.Add(type);
+            }
+        }
+
+        /// <summary>
+        /// Attaches the <see cref="ReactRootView" /> to the list of tracked root views
+        /// </summary>
+        /// <param name="rootView">The root view for the ReactJS app</param>
+        public override void AttachMeasuredRootView(ReactRootView rootView)
+        {
+            _attachedRootViews.Add(rootView);
+            
+            if (var != null)
+            {
+                AttachMeasuredRootViewToInstance(rootView, var.CatalystInstance);
+            }
+        }
+
+        /// <summary>
+        /// Detach given <see cref="rootView" /> from current catalyst instance. 
+        /// </summary>
+        /// <param name="rootView">The root view for the ReactJS app</param>
+        public override void DetachRootView(ReactRootView rootView)
+        {
+            if (_attachedRootViews.Remove(rootView))
+            {
+                if (var != null)
+                {
+                    DetachViewFromInstance(rootView, var.CatalystInstance);
+                }
+            }
+        }
+
+        private void DetachViewFromInstance(ReactRootView rootView, ICatalystInstance catalystInstance)
+        {
+            try
+            {
+                catalystInstance.GetJavaScriptModule<AppRegistry>()?.unmountApplicationComponentAtRootTag(rootView.TagId);
+            }
+            catch (InvalidOperationException ex)
+            {
+                throw new InvalidOperationException("Unable to load AppRegistry JS module. Error message: " + ex.Message);
```

Comment:
Not sure why you bother catching here to rethrow the same exception type.  At the very least, instantiate the new exception with the old one as an inner exception.